_posts/2016-12-26-controlling-access-with-proxy-pattern.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: post
+title: Controlling Access With Proxy Pattern
+author: ilu
+---
+
+![Proxy]({{ site.baseurl }}/assets/castle.jpg)
+
+## Introduction
+
+One of the most useful and versatile design patterns is the classic Proxy.
+
+## Proxy concept
+
+Basically it wraps an object and acts as a proxy between the calls to the object. At first thought it might not sound so useful but this simple concept is a great solution to many problems.
+
+![Proxy class diagram]({{ site.baseurl }}/assets/proxy-concept.png)
+
+In the class diagram we see that the client depends only on the interface so it can as well use the proxy instead of the real subject. When the proxy object is called it does its thing and eventually forwards the call to the real subject.
+
```

Comment:
- Static and Dynamic proxies can be one of the point to highlight here. The example that we are presenting is utilising the Static proxy where the proxy is pre built. 

After my work on Dynamic Proxy example
- After I complete the work on Dynamic proxy example which I was working on we can add a blog post specific to dynamic proxy as well.
- Comparison between static and dynamic can also be added once we have example of both types of proxies.